src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/ChimericAlignment.java

```diff
@@ -144,14 +142,20 @@ public ChimericAlignment(final AlignmentInterval intervalWithLowerCoordOnContig,
      * If the alignment region passes the above two filters and the next alignment region could be treated as potential inserted sequence,
      * note down the mapping & alignment information of that region and skip it
      * @param alignedContig          made of (sorted {alignmentIntervals}, sequence) of a potentially-signalling locally-assembled contig
-     * @param uniqueRefSpanThreshold for an alignment interval to be used to construct a ChimericAlignment,
-     *                               how long a unique--i.e. the same ref span is not covered by other alignment intervals--alignment on the reference must it have
      * @param referenceDictionary    reference sequence dictionary
+     * @param filterAlignmentByMqOrLength
```

Comment:
A refactoring idea for maybe improving this code, but don't treat this as a mandatory change for the PR if you dont' think it's useful: What if you passed in a list of filters to be applied instead of having different flags for each filtering type? Each filter object could be a predicate that takes an alignment interval and decides whether to filter it. That would simplify a lot of the if statements in this method.

Reply:
Thanks! This is nice. I'll make it this way, but in the future because:

currently, the filtering is applied non-homogeneously&mdash;the first several alignments are filtered based on MQ only and later ones are filtered based on alignment length and MQ. To introduce as less logic changes possible to master pipeline right now, I'll leave it as it is right now, and bring in the suggested more principled way when the alternative path is merged to master.